src/inputs/ObjectSearchCreate.tsx

```diff
@@ -69,21 +76,39 @@ class ObjectSearchCreate extends Component<IObjectSearchCreateProps> {
     return this.props.fieldConfig as IFieldConfigObjectSearchCreate;
   }
 
+  private get createFields(): IFieldConfig[] {
+    return getFieldSetFields(this.fieldConfig.createFields).map(createField => fillInFieldConfig(createField));
+  }
+
   private async onAddNew(search: string) {
     const { onAddNewToggle, formManager, fieldConfig } = this.injected;
     this.search = search;
 
-    formManager.form.setFieldsValue({ [fieldConfig.field]: {} }, () => {
-      this.isAddingNew.setTrue();
-      if (onAddNewToggle) {
-        onAddNewToggle(true);
-      }
-    });
+    const createFieldsToDefaults: FieldData[] = this.createFields.map(createField => ({
+        name: createFieldAntName(fieldConfig, createField),
+        value: formManager.getDefaultValue(createField),
+      })),
+      mainFieldAsEmptyObject = { name: fieldConfig.name, value: {} };
+
+    formManager.form.setFields([mainFieldAsEmptyObject, ...createFieldsToDefaults]);
+
+    this.isAddingNew.setTrue();
+    if (onAddNewToggle) {
+      onAddNewToggle(true);
+    }
   }
 
   private async onSearch() {
-    const { onAddNewToggle, formManager, id, fieldConfig } = this.injected;
-    formManager.form.setFieldsValue({ [id]: formManager.getDefaultValue(fieldConfig) });
+    const { onAddNewToggle, formManager, fieldConfig } = this.injected;
+
+    const createFieldsToUndefined: FieldData[] = this.createFields.map(createField => ({
```

Comment:
since a similar pattern is used above in `createFieldsToDefaults` could this map function be extracted?